### DISCUSSION TITLE
Typings for custom properties in AsyncHook Context

### DISCUSSION THREAD

**Discussion by NachtRitter** [TypedProject/ts-express-decorators#2083] [Q&A] [answered]
Hey @Romakita !

I'm faced with some problems of typings for my custom properties in the AsyncHook Context.
How can I properly augment types of PlatformContext?

I have middleware:
```ts
import { Middleware, MiddlewareMethods, PlatformContext } from '@tsed/common';

import { MyFooBar } from './MyFooBar';

@Middleware()
export class MyFooBarMiddleware implements MiddlewareMethods {
  use(@Context() $ctx: PlatformContext) {
    $ctx.set('foo', new MyFooBar());
  }
}
```
and I use context in service:

```ts
import { PlatformContext, Service } from '@tsed/common`;
import { InjectContext } from '@tsed/di';

@Service()
export class MyFooBarService {
  @InjectContext()
  context: PlatformContext;

  myMethod() {
    const bar = this.context.foo.bar();
  }
}
```

so, when I use property from context I'd like to have typings for it.

If it was Context, I would add my custom property to interface like 

```ts
declare global {
  namespace TsED {
    interface Context {
      foo: MyFooBar;
    }
  }
}
```

but I inject PlatformContext, so it doesn't work.
In code of framework I see that you don't have separate interface for PlatfromContext and you just make: 
```ts
declare global {
    namespace TsED {
        interface Context extends PlatformContext {
        }
    }
}
```
Could you please give some pieces of advice in which way I can solve my issue?
I was thinking about creation of my custom interface that will be extended by PlatformContext, but maybe you have better solution.

P.S. I use the latest 6.x version of framework

**Comment by Romakita** [comment, marked as answer]
Hello @NachtRitter 
You shouldn't do that :)

this exemple is wrong:

```typescript
import { PlatformContext, Service } from '@tsed/common`;
import { InjectContext } from '@tsed/di';

@Service()
export class MyFooBarService {
  @InjectContext()
  context: PlatformContext;

  myMethod() {
    const bar = this.context.foo.bar();
  }
}
```

It should be:

```typescript
import { PlatformContext, Service } from '@tsed/common`;
import { InjectContext } from '@tsed/di';

@Service()
export class MyFooBarService {
  @InjectContext()
  context: PlatformContext;

  myMethod() {
    const bar = this.context.get<MyFooBar>('foo').bar();
  }
}
```

Maybe my example answer you ;)

See you
Romain

**Reply by NachtRitter** [reply]
Huh, totally missed that in source code)
Now that makes sense.
Thank you!
